We.Sell.Bread.API.Integration.Tests/Tests/CustomerControllerTests.cs

```diff
@@ -1,78 +1,211 @@
-﻿using System.Net;
-using Microsoft.AspNetCore.Mvc.Testing;
+﻿using Castle.Core.Resource;
+using System.Net.Mail;
+using System.Net.NetworkInformation;
+using We.Sell.Bread.API.Integration.Tests.Helpers;
+using We.Sell.Bread.Core.DTOs.Customer;
 
 namespace We.Sell.Bread.API.Integration.Tests.Tests
 {
-    public class CustomerControllerTests
+    public class CustomerControllerTests(WebApplicationFactory<Program> factory) : IClassFixture<WebApplicationFactory<Program>>
     {
-        private const string _baseUrl = "https://localhost";
-        private readonly HttpClient _httpClient;
-        public CustomerControllerTests() 
+        private readonly HttpClient _httpClient = factory.CreateClient();
+
+        [Fact]
+        public async Task GivenCustomerControllerWhenCheckingForServerHealthReturnOkStatusAsync()
         {
-            var webApplicationFactory = new WebApplicationFactory<Program>()
-                .WithWebHostBuilder(builder => 
-                {
-                    builder.ConfigureServices(services => { });
-                });
+            var response = await _httpClient.GetAsync($"/Customer/HealthCheck");
 
-            _httpClient = webApplicationFactory.CreateClient(new() {BaseAddress = new Uri(_baseUrl)});
+            response.IsSuccessStatusCode.Should().BeTrue();
+            response.StatusCode.Should().Be(HttpStatusCode.OK);
         }
 
         [Fact]
-        public async Task GivenCorrectCustomerIdWhenRetrievingCustomerReturnOkStatusCodeAndCustomerDetails()
+        public async Task GivenCorrectCustomerIdWhenRetrievingCustomerReturnOkStatusCodeAsync()
         {
-            //var customerId = "ad3d2d6c-02fd-48c9-a287-fa33aa197053";
+            var customerId = "d1a993e8-398a-4ada-9035-b16d280ba987";
```

Comment:
Because we are using one id on all the methods that pass in a correct id, for easy future maintenance, Let either create a const variable and make use of that variable or create a test data file and have these id's in that file.

Reply:
Fixed